sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -1443,4 +1443,22 @@ public String getSnapshotId() {
     public boolean isSnapshot() {
         return this.snapshot != null;
     }
+
+    /**
+     * Get the blob name.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobAsyncClient.getName}
+     *
+     * @return The name of the blob.
+     */
+    public String getName() {
+        try {
+            return URLParser.parse(new URL(this.azureBlobStorage.getUrl())).getBlobName();
```

Comment:
Does this throw a `MalformedURLException`?

Reply:
`new URL(String)` throws a MalformedURLException

Reply:
Move to URLParser